pdf-service/Controllers/AuthenticationController.cs

```diff
@@ -45,5 +47,48 @@ public string GetToken(string secret)
 
             return result;
         }
+
+        [HttpGet("redirect/{secret}")]
+        [AllowAnonymous]
+        // response_type=token&client_id=&redirect_uri=https%3A%2F%2Flocalhost%3A5001%2Fswagger%2Foauth2-redirect.html&state=VHVlIERlYyAwOCAyMDIwIDE0OjE1OjE2IEdNVC0wODAwIChQYWNpZmljIFN0YW5kYXJkIFRpbWUp
+        public ActionResult SetToken([FromRoute] string secret, string redirect_uri, string response_type, string state)
+        {
+            string result = "Invalid secret.";
+            string token = "";
+            string expires_in = "";
+            string configuredSecret = Configuration["JWT_TOKEN_KEY"];
+            if (configuredSecret.Equals(secret))
+            {
+                var key = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(Configuration["JWT_TOKEN_KEY"]));
+                var creds = new SigningCredentials(key, SecurityAlgorithms.HmacSha256);
+
+                var claims = new[] {
+                    new Claim(JwtRegisteredClaimNames.Jti, Guid.NewGuid().ToString())
+                };
+
+
+                var jwtSecurityToken = new JwtSecurityToken(
+                    Configuration["JWT_VALID_ISSUER"],
+                    Configuration["JWT_VALID_ISSUER"],
+                    expires: DateTime.UtcNow.AddDays(1),
+                    signingCredentials: creds
+                );
+                token = new JwtSecurityTokenHandler().WriteToken(jwtSecurityToken);
+                result = new JwtSecurityTokenHandler().WriteToken(jwtSecurityToken) + " Expires:" + jwtSecurityToken.ValidTo.ToShortTimeString();
+
+            }
+
+            if (!string.IsNullOrEmpty(redirect_uri))
+            {
+                Response.Headers.Add("Authorization", $"Bearer {result}");
+                string tokenEncoded = UrlEncoder.Default.Encode(token);
+                string redirect = redirect_uri +
+                                  $"#state={state}&token_type=Bearer&access_token={tokenEncoded}&id_token={tokenEncoded}&access_token={tokenEncoded}&expires_in=99999";
```

Comment:
This looks good to me but what's the need for having the same `access_token` query param twice in the URL?

Reply:
good catch